app/demo-loop/LDemoIO.root.cc

```diff
@@ -7,15 +7,64 @@
 //---------------------------------------------------------------------------//
 #include "LDemoIO.hh"
 
+#include <string>
+#include <vector>
 #include <TBranch.h>
 #include <TFile.h>
 #include <TTree.h>
 
+#include "celeritas/global/ActionRegistry.hh"
+
 namespace demo_loop
 {
 //---------------------------------------------------------------------------//
+/*!
+ * Store CoreParams data to the ROOT MC truth output file.
+ * Called by `to_root(...)`.
+ *
+ * \note
+ * Currently only storing the action labels so their IDs can be
+ * identified. If other parameters are needed for future
+ * debugging/analyses, this function can easily be expanded.
+ */
+void store_core_params(std::shared_ptr<celeritas::RootFileManager>& root_manager,
+                       celeritas::CoreParams core_params)
```

Comment:
Use just a mutable reference to the RFM here (you're not modifying the pointer but the data it's pointing to). Use a `CP const&` for the core params instead of copying them.